src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -114,12 +131,16 @@ void ucc_tl_ucp_allgather_knomial_progress(ucc_coll_task_t *coll_task)
                     continue;
                 }
             }
-            UCPCHECK_GOTO(ucc_tl_ucp_send_nb(sbuf, local_seg_count * dt_size,
+            UCPCHECK_GOTO(ucc_tl_ucp_send_nb_with_mem(sbuf, local_seg_count * dt_size,
                                              mem_type,
                                              ucc_ep_map_eval(task->subset.map,
                                              INV_VRANK(peer, broot, size)),
-                                             team, task),
+                                             team, task, mh_list[count_mh++]),
                           task, out);
+            if (count_mh >= max_count){
```

Comment:
there is a function ucc_assert for that look what its doing

Reply:
why u chose to out that check here?

Reply:
should I just raise an error maybe ? How ?

Reply:
if I use ucc_assert, I can't add my print right ?